pkg/apis/pipeline/v1alpha1/type.go

```diff
@@ -46,10 +46,39 @@ type PipelineSpec struct {
 	// The tasks will be executed in the order they are listed.
 	Tasks []PipelineTask `json:"tasks"`
 
-	// SharedWorkspace is the name of the PVC. If not specified, a PVC with the Pipeline's name as prefix will be created by default.
+	// SharedWorkspace is the config of the PVC where task using
+	// The PersistentVolumeClaim with this config will be created for each pipeline execution
+	// it allows the user to specify e.g. size and StorageClass for the volume.
 	// If not set, Kurator will create a PVC named Pipeline.name using default config
 	// +optional
-	SharedWorkspace *string `json:"sharedWorkspace,omitempty"`
+	SharedWorkspace *VolumeClaimTemplate `json:"sharedWorkspace,omitempty"`
```

Comment:
I did not realize: isn't this a PVC spec. why do we redefine a new one?

Reply:
Using a pre-created PVC (Persistent Volume Claim) in a pipeline then references it name is not recommended, as PVCs are meant to be linked with each specific pipeline execution instance, not the pipeline itself.